std/assembly/util/uri.ts

```diff
@@ -0,0 +1,99 @@
+// References:
+// https://github.com/brianmario/escape_utils/blob/master/ext/escape_utils/houdini_uri_u.c
+
+import { CharCode } from "./string";
+
+// @ts-ignore: decorator
+@lazy export const URI_SAFE = memory.data<u8>([
```

Comment:
With only zeroes and ones, could these arrays become more like bit flags to save some space (16 instead of 127 bytes)? :)

Reply:
The main goal of this array is speedup verification. For preferred size version (like -Oz) just better use logical range checks

Reply:
Hmm, but wouldn't this compile to branchless bitwise logic on two `i64`s in memory anyhow, so wouldn't hurt much?

Reply:
@dcodeIO I experimented with different approach and I guess if we invert current lookup table it will be much smaller and still fassted "bit-packed" table:
https://webassembly.studio/?f=jkjghbmygxr

1) Full table (current) - 207 bytes
2) Optimized Full table - 158 bytes
3) Bit packed table - 118 bytes

WDYT?

Reply:
It could be even more optimial. See `checkFullOpt2`:

4. Optimized Full table 2 + code - 126 bytes

https://webassembly.studio/?f=hqlv5gpr1kk